jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -664,6 +664,9 @@ def calculate_fit(self, add_data=True):
         fitted spectrum/cube
         residuals (if ``residuals_calculate`` is set to ``True``)
         """
+        if not self.spectral_subset_valid:
+            raise ValueError("spectral subset is outside data range")
```

Comment:
This only ever happens if ignoring the UI warnings and calling from the API directly... although that also means the user likely isn't staring at the viewer where it's visually obvious.  I'm not convinced ranges will help though and am guessing the more likely action would be to change one selection or the other... so do you think it would help to be `"spectral subset '{self.spectral_subset.selected}' is outside data range of '{self.dataset.selected}'"`?

Reply:
ok, I added both.  Let me know what you think!